python/dask_cudf/dask_cudf/io/parquet.py

```diff
@@ -142,23 +142,32 @@ def read_partition(
             if partitions is None:
                 raise ValueError("Must pass partition sets")
 
+            categorical_partitions = kwargs.pop("categorical_partitions", True)
             for i, (name, index2) in enumerate(partition_keys):
-
-                # Build the column from `codes` directly
-                # (since the category is often a larger dtype)
-                codes = (
-                    as_column(partitions[i].keys.index(index2))
-                    .as_frame()
-                    .repeat(len(df))
-                    ._data[None]
-                )
-                df[name] = build_categorical_column(
-                    categories=partitions[i].keys,
-                    codes=codes,
-                    size=codes.size,
-                    offset=codes.offset,
-                    ordered=False,
-                )
+                if categorical_partitions:
+                    # Build the column from `codes` directly
+                    # (since the category is often a larger dtype)
+                    codes = (
+                        as_column(partitions[i].keys.index(index2))
+                        .as_frame()
+                        .repeat(len(df))
+                        ._data[None]
+                    )
+                    df[name] = build_categorical_column(
+                        categories=partitions[i].keys,
+                        codes=codes,
+                        size=codes.size,
+                        offset=codes.offset,
+                        ordered=False,
+                    )
+                else:
+                    # Repeat index2 for every element of the new column
```

Comment:
do you think it's worth adding a note that round tripping dtypes won't work.   For example, Int64 columns can be int32 after reading